tests/modify.js

```diff
@@ -50,19 +49,39 @@ describe('modify', () => {
         const h1 = hamt.empty
             .modifyHash(0, 'a', () => 3)
             .modifyHash(0, 'b', () => 5);
-    
+
         const h3 = h1.modifyHash(0, 'a', x => x * 2);
         assert.strictEqual(6, h3.getHash(0, 'a'));
         assert.strictEqual(5, h3.getHash(0, 'b'));
-    
+
         const h4 = h3.modifyHash(0, 'b', x => x * 2);
         assert.strictEqual(6, h4.getHash(0, 'a'));
         assert.strictEqual(10, h4.getHash(0, 'b'));
-    
+
         // Non existant
         const h5 = h4.modifyHash(0, 'c', _ => 100);
         assert.strictEqual(6, h5.getHash(0, 'a'));
         assert.strictEqual(10, h5.getHash(0, 'b'));
         assert.strictEqual(100, h5.getHash(0, 'c'));
     });
+
+    it('should provide defaultValue to `f` if entry does not exist on empty map', () => {
+        // TODO: complete
+    });
+
+    it('should provide defaultValue to `f` if entry does not exist on non-empty map', () => {
+
+        const h = hamt.set('a', 3, hamt.empty);
+
+        const NOT_SET = {};
+
+        const transform = (x) => {
+
+            assert.strictEqual(x, NOT_SET);
+
+            return 'b';
+        };
+
+        const h1 = hamt.modify(transform, 'b', NOT_SET, h);
```

Comment:
Not sure this is testing the correct thing. Let's check that we have the same understanding of the semantics of `defaultValue`.

In the case where `defaultValue` is provided, then `f` is only called for modifications operations. For insertions, `f` is not called and instead `defaultValue` is used. In a case without `defaultValue`, `f` is called for both insertions and modifications (this is the current behavior and cannot change since this is targeting a minor release).

The problem here is the `transform` is never called in my understanding, do the assertion will never run. Please let me know if you had different semantics in mind. Updating the test to make sure `b` is now set to `NOT_SET` should do the trick.